### ISSUE TITLE
Diversify space background in order to make it more interesting

### ISSUE THREAD

**Issue by Tadrix** [endless-sky/endless-sky#4420]
For starters, I'm apologising in advance for the text wall. I've yet to learn to express myself more briefly in English.

**Is your feature request related to a problem? Please describe.**
Honestly, flying for many years through the galaxy, it gets visually boring. The systems visually differ only with their stars, and the asteroid density. The eye learns to recognise tiling patterns on the fly, and it's a bad thing. Put bluntly, the space background is dull and boring. It might be somewhat realistic, as far as realism in this game goes, but after seeing [photographs](https://duckduckgo.com/?q=night+sky&iax=images&ia=images) (yes, I know they had a long exposition time), one would think the game can do better. We're playing for fun, after all, not for a realistic simulation of routine, repetitive tedium.

Even the game advertises itself a bit differently. Here's the endless-sky's avatar:

![Avatar 11450262](https://avatars2.githubusercontent.com/u/11450262)

Looking at it for a long time would probably be tiring (I've always thought that in SPAZ it was overdone, and seeing a black sky was a relief from being immersed in oversaturated brightness; in Homeworld, however, it might've been done just right), but it's a nice alternative to the usual blue haze. The interface features something akin to the red haze too:

![Hail Panel](https://raw.githubusercontent.com/endless-sky/endless-sky-high-dpi/master/images/ui/hail%20panel%402x.png)

Seeing a system with a [nova](https://github.com/endless-sky/endless-sky/blob/master/images/star/nova%2B.jpg), a [Wolf-Rayet star](https://github.com/endless-sky/endless-sky/blob/master/images/star/wr.png), or a [black hole](https://github.com/endless-sky/endless-sky/blob/master/images/star/black-hole%2B.jpg) in it is great. *Whoa!* The background is gorgeous there. It's a shame you as a player don't get to stay around for long, because such systems understandably are completely devoid of any business to be had, so people don't tend to linger there.

Technically speaking, every star has its own spectral characteristics (one of the reasons they are shown coloured differently). Everything would look a bit differently depending on what star is present. Instead, everything looks the same (I'm not sure that can be corrected, and if any filters are a good idea, but it certainly compounds the problem).

**Related Issue Links**
#2488, although I don't think having colours based on what faction inhabits what part of galaxy would be a good idea. It should depend on the part of the galaxy, and the local star.

**Possible solutions**
In short, making the galaxy a little more visually interesting place. I'm not a designer, so I don't have ready solutions.

The space might be brighter, the closer the system is to the galaxy core (hopefully, not so bright one would want to gouge one's eyes out). Its colour/lighting might slightly vary depending on what star is present. Or maybe having bigger start sprites with features around them like in aforementioned examples would help (for example, perhaps the Sol system might use an asteroid belt in the background between Mars and Jupiter). Depends on what we can, and what changes would be acceptable. Maybe there are even better ways of accomplishing that.

**Comment by Zitchas** [comment]
Sounds good. I'd love to have a larger variety of backgrounds and systems. 

It'd be even more interesting if these backgrounds could tie into the system hazards I've been hoping for.

**Comment by Pointedstick** [comment]
I have to say I agree.

The current background does have a subtle haze effect, but I think it could be both more obvious and also more varied across different parts of the galaxy too.

Re: the balance between visual flair and realism, I think it's important to not to make the mistake that this is a "hard sci-fi" game. Even though it lacks space magic and strives to be internally consistent, Endless Sky is in almost no way a hard sci-fi game, and as such, I think we should not reject proposed visual enhancements simply on the basis that they're not strictly speaking scientifically accurate.

**Comment by Lineth** [comment]
> Even though it lacks space magic 

Ehh, just as a note, the game may not explicitly _say_ it, but paranormal phenomena are implied very heavily at several points whenever the Pug or Drak are the subject.

**Comment by Brick63** [comment]
Any advanced enough science is indiscernible from magic

**Comment by ChamEV** [comment]
Accidentally did this a while back: https://cdn.discordapp.com/attachments/266345072554016768/594085799893532673/unknown.png way too bright but it certainly made the menu look cooler.

**Comment by Tadrix** [comment]
People tend to notice the haze sprite issue (as I named it right in the topic), but overlook the other possible improvements (like background-including star sprites, mentioned in the OP as well). Any thoughts on them?

**Comment by Amazinite** [comment]
Perhaps the brightness and solar wind strength of the stars in the system could be used.
The brighter the system, the brighter the haze (meaning that a direct change to the haze images like #4434 wouldn't be in order, but a change to the game's code to do something like change the opacity of the haze depending on the system brightness).
The stronger the solar wind, the stronger some sort of effect in the system could be. Perhaps a subtle "wind" of particles streaming from the stars in the system? Then the stronger the wind, the more frequently the particles are emitted. 

WRT the idea of the haze getting brighter the closer you are to the galactic core, not so sure on that. How would each system know its distance from the core? How would that work if we added other galaxies? Maybe each system would have an attributed that is the coordinates of it's galaxy's core. For example, SagA* is at the position 112 22 in game, so Sol would have a new attribute like follows.
```
system Sol
	pos -400 100
	core 112 22
	government Republic
	habitable 1080
```
Then just call upon Pythagoras to find the distance from the core and have some algorithm determine the haze brightness from that.  Since the brightness of any haze in reality would be nearly entirely effected by the local star(s) though, perhaps the distance from the core would just effect the number of specks in the background, as those are suppose to be distant stars (despite the excessive parallax).

**Comment by Tadrix** [comment]
> WRT the special stellar objects, not sure what they'd do any different from the above ideas. They're already highly different on their own, so no idea what extra could be done.

My point was that the examples mentioned depict not only the star, but also a part of the system. Some of the ordinary stars could have something similar as well—their own haze, or a halo, or just brighter sky (rays? as in, light reflecting on dust particles), something that makes the sky in immediate vicinity of the star different. But it probably wouldn't work for binary systems or those with planets very close.

The rest sounds great to me.

**Comment by MCOfficer** [comment]
to throw in another suggestion: i played around with post processing (specifically, [reshade](https://reshade.me/)), and i would love ES to have just a tad of bloom.

A small comparison: a bit of bloom on the right, standard settings on the left.
![grafik](https://user-images.githubusercontent.com/22377202/61390920-f4beb380-a8bb-11e9-8306-06e00e879912.png)

As you can see, the bloom is only truly visible around the bright parts of the image (cause well, that's how bloom works). so it isn't the one solution for this issue, but it could go a long way of enhancing any solution that is currently on the table.

**Comment by Amazinite** [comment]
When it comes to making the visual effects more eye appealing, it's probably worth mentioning animated stars (#1517 and #1524).

>I really want the space backdrop to feel somewhat sparse and empty, and not to be visually distracting. [source](https://github.com/endless-sky/endless-sky/issues/1517#issuecomment-244567376)

**Comment by Pointedstick** [comment]
I generally agreed with MZ regarding the animated stars and planets ideas, but I think a vit more visual flair can work if done tastefully. For example, rotating space stations, as long as they rotate very slowly. I also think making the background haze more obvious is in the same category as long as we don't go overboard. There's always a fine line between "gaudy/distracting" and "too subtle", but I think we can thread that needle.

**Comment by Zitchas** [comment]
Some variation would be good, I think. Some thoughts:

Brightness haze (such as the one Amazinite suggested tied to the solar wind) should probably decrease along with it, so that the farther one gets, the darker and emptier space becomes.

On the flip side, haze such as the Ember Waste nova-haze should probably remain fairly constant everywhere (or at least, vary randomly as opposed to at fixed distances from stars).

Having some kind of variables allowing us to set different levels of haze in different systems (basically, an opacity or transpraency percent value attached to the current haze entry in map.txt) would allow for a lot of variation right there.

Another idea is allowing multiple hazes with variable intensities. Thus with a set of three or four hazes (a standard color set), we could allow for background haze of almost any colour or density.

**Comment by Amazinite** [comment]
#4986 added more hazes to the game, and #5828 seeks to add more unique stars. There's also been talk of allowing you to alter the number of stars in the far background, something that was also discussed in this issue. So if this issue isn't resolved yet, I think we're perhaps close.

**Comment by TheGiraffe3** [comment]
It's now possible to alter the star count using the `"star density"` attribute.

**Comment by Amazinite** [comment]
I think it's fair to say that we've spiced up the background a sufficient amount.

> It's now possible to alter the star count using the `"star density"` attribute.

Ref #7765
